src/routing/approved.rs

```diff
@@ -665,7 +668,38 @@ impl Approved {
                 self.handle_join_request(msg.src().to_node_peer(sender)?, *join_request.clone())
             }
             Variant::UserMessage(content) => {
-                self.handle_user_message(msg.src().src_location(), *msg.dst(), content.clone())
+                if msg.dst() == &DstLocation::AccumulatingNode(self.node().name()) {
+                    if let SrcAuthority::BlsShare { proof_share, .. } = msg.src() {
+                        let signed_bytes = bincode::serialize(&msg.signable_view())?;
+                        match self
+                            .message_accumulator
+                            .add(&signed_bytes, proof_share.clone())
+                        {
+                            Ok(proof) => {
+                                trace!("Successfully accumulated message: {:?}", msg);
+                                let key = msg.proof_chain_last_key()?;
+                                if key.verify(&proof.signature, signed_bytes) {
+                                    self.handle_user_message(
+                                        msg.src().src_location(),
+                                        *msg.dst(),
+                                        content.clone(),
+                                    )
+                                } else {
+                                    trace!("Aggregated signature is invalid. Handling message {:?} skipped", msg);
+                                    Ok(vec![])
+                                }
+                            }
+                            Err(err) => {
```

Comment:
The signature aggregator also returns `Error::NotEnoughShares` which is not really an error so should probably not be logged to avoid spamming the log? See also `Approved::handle_vote`.